bfvmm/include/intrinsics/x86/intel/apic/x2apic.h

```diff
@@ -2134,7 +2223,95 @@ namespace x2apic
 /// mode. It is marked final because it is intended to interact
 /// directly with x2apic hardware.
 ///
-struct EXPORT_X2APIC x2apic_control final : public lapic_control {
+struct EXPORT_X2APIC x2apic_control final : public lapic_control
+{
+    //
+    // Check if guest physical address is an APIC register and the desired
+    // read / write operation is allowed.
+    //
+    // @return offset if supplied address maps to a valid register and the
+    //    operation is allowed.
+    // @return -1 if the supplied address doesn't map to a valid register or the
+    //    operation is not allowed.
+    //
+    // @param addr - guest physical address of desired register
+    // @param op - the desired operation (read / write)
+    //
+    int validate_gpa_op(uintptr_t addr, reg_op op) noexcept override
```

Comment:
alias uintptr_t as gpa_type in the class definition and use that here. Also mark the arguments as consts